token/common.tact

```diff
@@ -93,3 +93,33 @@ struct BurnNotification {
 
     @derive impl Serialize {}
 }
+
+contract interface JettonOwner {
+    msg receive_notification(prefix: 0x7362d09c, notification: TransferNotification);
```

Comment:
I guess we need to clearly separate requirements _for dispatch_ and requirements _after dispatch_.

If we want to say "this method is for prefix 0x734c12a4" - that's a for-dispatch requirement. Similarly for case like "this method catches all bounced messages".

Maybe there is a very limited set of dispatch conditions (some flags and prefix), while all the rest are after-dispatch requirements that could also be checked in runtime with hard failure.

Reply:
Also: things like "bounced: true" are not a part of the interface at all. But "bounceable: true" could very well be (checked in runtime by receiver and in comptime by the sender).

Reply:
Yeah, only requirements _for dispatch_ should be displayed in the interface. These requirements are:
1. Prefix.
2. Sender interface (but this won't be checked in the runtime, of course).
3. Amount of coins appended to the message.
4. `bounceable` flag.

Reply:
Maybe we could say `bounce transfer#1234(...)` to catch bounced message for this prefix. And `bounce default()` to catch all bounced.

Reply:
```
internal <name>|default
external <name>|default
bounced <name>|default
```

...and, fasten your seatbelts:

```
internal+bounced <name>(...)
```

Reply:
Only internal message can be bounced, so we do not need `internal+bounced` keyword.

Reply:
i mean:
* `internal` = internal non-bounced
* `bounced` = internal bounced
* `internal+bounced` = internal bounced=any, because I want to handle both cases in one handler.

The first two are good defaults for most contracts. Most contracts want to process internal non-bounced msgs and ignore or bulk-process all the bounced ones in one handler.